lib/ecto.ex

```diff
@@ -570,6 +570,26 @@ defmodule Ecto do
   def assoc_loaded?(%_{}), do: true
   def assoc_loaded?(nil), do: true
 
+  @doc """
+  Reset fields in a struct to their default values.
+
+  ## Examples
+
+      iex> post = post |> Repo.preload(:author)
+      %Post{title: "hello world", author: %Author{}}
+      iex> Ecto.reset_fields(post, [:title, :author])
+      %Post{
+        title: "default title",
+        author: #Ecto.Association.NotLoaded<association :author is not loaded>
+      }
+
+  """
+  def reset_fields(%{__struct__: schema} = struct, fields) do
```

Comment:
```suggestion
  @spec reset_fields(Ecto.Schema.t(), list()) :: Ecto.Schema.t()
  def reset_fields(%{__struct__: schema} = struct, fields) do
```

Sorry my bad, I missed this. It would be nice to have the type specs.

Reply:
Thanks for providing the code snippet. It helps a lot.

Reply:
And I rebased it on the first commit.